### DISCUSSION TITLE
Hidden content filter misbehaving? Problem with 'Sex'

### DISCUSSION THREAD

**Discussion by Aivean** [community/community#6772] [Copilot Conversations] [answered]
I'm playing with a [toy datascience problem on Kaggle](https://www.kaggle.com/c/titanic/overview) using Copilot. 

Overall, it appears immensely useful for quick data exploration and transformation. However, I've encountered a weird quirk. 

There is a column named 'Sex' in the training data:
![image](https://user-images.githubusercontent.com/2865203/138939837-4af8ceab-4faa-4cf6-824b-d4e20341bdea.png)

As soon as Copilot sees the word 'Sex' in the code it stops working. E.g., when I add this line to my notebook,
```
# convert 'Sex' column into binary representation
train_processed['Sex'] = train_processed['Sex'].apply(lambda x: 1 if x == 'male' else 0)
```
Copilot completions stop working for all lines following the aforementioned code (they still work for the preceding lines).

Is this some kind of content filter misbehaving? Is this behavior coming from OpenAI, or is it something on the Copilot side? Is it supposed to work like this?

---

**UPD**

I tried to create a minimal reproducible test for this weirdness. Here is the whole content of the file (`test11.ipynb`):

### Cases when completion stops working:
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'female' else 0)
```
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'male' else 0)
```
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'boy' else 0)
```
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'girl' else 0)
```

### Cases when completion works fine:
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'femal' else 0)
```
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'human' else 0)
```
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'red' else 0)
```
```
#%%
train_processed['Test'] = train_processed['Test'].apply(lambda x: 1 if x == 'person' else 0)
```

Is this some kind of gender discrimination prevention mechanism or am I just seeing things?

**Comment by Aivean** [comment, marked as answer]
Ok, after some research, I discovered, that there is indeed a [content filter](https://copilot.github.com/#faq-does-github-copilot-produce-offensive-outputs), that, at this point, appears to be pretty rudimentary (as in, it produces false negatives, while it can be easily circumvented by string concatenation).

I just hope that this filter is improved in the future, so innocuous words, such as 'boy' won't break copilot altogether.

**Reply by asonnleitner** [reply]
Facing the same issue with everything related to gender.

**Reply by subssn21** [reply]
This is a really annoying issue. I was working great this morning, on some work with a patient class, and I have some fields called legal_sex and sex_at_birth, As soon as I got to those fields CoPilot just stopped working.
